lib/giraph/remote/response.rb

```diff
@@ -0,0 +1,9 @@
+module Giraph
+  module Remote
+    # Dummy Hash-wrapper to enable precise `instance_of?` checks
+    # Allows us to differentiate JSON responses that are
+    # coming from remote GraphQL interface vs regular Hash objects
+    class Response < Hash
```

Comment:
Interesting - I didn't realize that the recursive nesting was required. It would be good to add a unit test to explain this.